### PR TITLE
Add letters() and syllables() for FPGroupElem

### PR DESCRIPTION
**Pull request by fingolfin**
- Add letters() and syllables() for FPGroupElem
- Add some type assertions to GAP code

### CODE REVIEW

File: src/Groups/GAPGroups.jl [hunk 1]
```diff
@@ -1590,6 +1592,83 @@ function _map_word_syllable(vi::Pair{Int, Int}, genimgs::Vector, genimgs_inv::Ve
 end
 
 
+@doc Markdown.doc"""
+    syllables(g::FPGroupElem)
+
+Return the syllables of `g` as a list of pairs `(gen,exp)` where
```
Comment (thofma) [inline src/Groups/GAPGroups.jl]:
```suggestion
Return the syllables of `g` as a list of pairs `gen => exp` where
```
Maybe use the pair syntax?

File: src/Groups/GAPGroups.jl [hunk 2]
```diff
@@ -1590,6 +1592,83 @@ function _map_word_syllable(vi::Pair{Int, Int}, genimgs::Vector, genimgs_inv::Ve
...
+`gen` is the index of a generator and `exp` is an exponent. 
+
+# Examples
+```jldoctest
+julia> F = free_group(2);  F1 = gen(F, 1);  F2 = gen(F, 2);
```
Comment (thofma) [inline src/Groups/GAPGroups.jl]:
```suggestion
julia> F = free_group(2);  F1, F2 = gens(F);
```
? Feel free to dismiss

File: src/Groups/GAPGroups.jl [hunk 3]
```diff
@@ -1590,6 +1592,83 @@ function _map_word_syllable(vi::Pair{Int, Int}, genimgs::Vector, genimgs_inv::Ve
...
+@doc Markdown.doc"""
+    letters(g::FPGroupElem)
+
+Return the letters of `g` as a list of integers, each entry corresponding to
+a group generator. Inverses of the generators are represented by negative
+numbers.
+
+# Examples
+```jldoctest
+julia> F = free_group(2);  F1 = gen(F, 1);  F2 = gen(F, 2);
```
Comment (thofma) [inline src/Groups/GAPGroups.jl]:
```suggestion
julia> F = free_group(2);  F1, F2 = gens(F);
```

**Review by thofma — APPROVED** [review]
Awesome! I played around with it and it seems to work quite well.

**Review by ThomasBreuer — APPROVED** [review]
Looks good.

What happened to @fingolfin's idea that the exponents in the output of `syllables` can be `fmpz`s (perhaps even by default), and @thofma's remark that then one should be able to prescribe the type, analogous to the case of exponent vectors of polynomials?

**Comment by fingolfin** [comment]
I wanted to get something useful out *now* instead of waiting for a discussion on this (which in practice may not lead anywhere). Also, GAP is not able to deal with huge exponents in letter rep for obvious reasons, so no need for fmpz there. Since it is the default, dealing with large exponent, this seems OK for now. We can add a variant that allows prescribing a return type later on if there is demand for it or someone feels bored

**Comment by thofma** [comment]
(Tests are failing)

**Pull request merged**